runtime/doc/api-ui-events.txt

```diff
@@ -881,6 +882,24 @@ must handle.
 	    True if the message should be appeneded to the previous message,
 	    rather than started on a new line. Is set for |:echon|.
 
+	msg_id
+	    Unique identifier for the message in history. Is greater or equal
+	    to 1. 0 indicates msg_id was not assigned. When message of same id
```

Comment:
Always assigning/emitting an ID would be less surprising. Though I suppose this `0` conveys that no caller is aware of the msg->id mapping (and thus shouldn't be replaced in the future) unlike `nvim_echo` emitted messages? Though given https://github.com/neovim/neovim/pull/34846/files#r2299325332 we might be better off just always assigning an ID as suggested in https://github.com/neovim/neovim/issues/32537#issuecomment-2672600804:

> All messages (incl. non-progress) have an id."

Also `msg_` is redundant here. Better use `id`?

Reply:
autocmd events themselve has a id feid. ti becomes something like

```
{
id: ,
...
data: {
   id: ,
   ....
}
}
```

I thought that was confusing so the `msg_`